### PR TITLE
8268727: Remove unused slowpath locking method in OptoRuntime

### PR DESCRIPTION
**Pull request by miao-zheng**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8268727](https://bugs.openjdk.java.net/browse/JDK-8268727): Remove unused slowpath locking method in OptoRuntime

### Reviewers
 * [Dean Long](https://openjdk.java.net/census#dlong) (@dean-long - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4490/head:pull/4490` \
`$ git checkout pull/4490`

Update a local copy of the PR: \
`$ git checkout pull/4490` \
`$ git pull https://git.openjdk.java.net/jdk pull/4490/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4490`

View PR using the GUI difftool: \
`$ git pr show -t 4490`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4490.diff">https://git.openjdk.java.net/jdk/pull/4490.diff</a>

</details>

### CODE REVIEW

**Comment by miao-zheng** [comment]
Could someone help to review this change? Thanks.

**Comment by dholmes-ora** [comment]
Redirecting to the compiler team.

File: test/lib/sun/hotspot/WhiteBox.java [hunk 1]
```diff
@@ -310,7 +310,7 @@ public         void    makeMethodNotCompilable(Executable method, int compLevel,
     makeMethodNotCompilable0(method, compLevel, isOsr);
   }
   public        int     getMethodCompilationLevel(Executable method) {
-    return getMethodCompilationLevel(method, false /*not ost*/);
+    return getMethodCompilationLevel(method, false /*not osr*/);
```
Comment (zhengyu123) [inline test/lib/sun/hotspot/WhiteBox.java:313]: This change is unrelated, should be fixed in a separate CR?

**Review by zhengyu123 — CHANGES_REQUESTED** [review]

File: test/lib/sun/hotspot/WhiteBox.java (hunk 1 above)
Comment (miao-zheng) [reply test/lib/sun/hotspot/WhiteBox.java:313]: This is a small typo fix, can I merge it in this patch?

Comment (zhengyu123) [reply test/lib/sun/hotspot/WhiteBox.java:313]: I think the convention is "no"

Comment (miao-zheng) [reply test/lib/sun/hotspot/WhiteBox.java]: Got it. Thanks for your review. I have removed this change.

**Review by dean-long — APPROVED** [review]
Looks good, and trivial.  It's strange that some pre-commit tests failed on Windows.

**Comment by miao-zheng** [comment]
The same test cases fail in other Pull requests on windows,  for example: https://github.com/openjdk/jdk/pull/4534.
I think these test cases fail are random.

**Comment by dean-long** [comment]
I agree, test failures seem unrelated to this change.

**Comment by miao-zheng** [comment]
@dean-long @zhengyu123 Thanks for your review.
